cmd/main.go

```diff
@@ -31,3 +39,21 @@ func echoBadRequestHandler(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "query：%s\n", query)
 	}
 }
+
+func echoCheckIdHandler(w http.ResponseWriter, r *http.Request) {
+	validate := validator.New()
+
+	id := r.URL.Query().Get("id")
+	name := r.URL.Query().Get("name")
+	idInt, _ := strconv.Atoi(id)
+	query := checkId{
+		ID:   idInt,
+		Name: name,
+	}
+
+	if err := validate.Struct(query); err != nil {
+		http.Error(w, "Bad request", http.StatusBadRequest)
+	} else {
+		fmt.Fprintf(w, "ID：%v\nName:%v", idInt, name)
+	}
```

Comment:
こうかな。
なるべくelseは使わないほうが読みやすいし、流れが明確になります。
```suggestion
	if err := validate.Struct(query); err != nil {
		http.Error(w, "Bad request", http.StatusBadRequest)
		return
	}
	fmt.Fprintf(w, "ID：%v\nName:%v", idInt, name)
```

Reply:
returnで返せば良いんですね。修正しました！